mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -322,56 +322,190 @@ pub(crate) async fn block_header_utxo_loop<T: UtxoCommonOps>(
+                    sync_status_loop_handle.notify_on_temp_error(&err);
                     Timer::sleep(args.error_sleep).await;
                     continue;
                 };
 
                 error!(
-                    "Error {error:?} on retrieving latest {ticker} headers from rpc after {FETCH_BLOCK_HEADERS_ATTEMPTS} attempts"
+                    "Error {err:?} on retrieving latest {ticker} headers from rpc after {FETCH_BLOCK_HEADERS_ATTEMPTS} attempts"
                 );
                 // Todo: remove this electrum server and use another in this case since the headers from this server can't be retrieved
-                sync_status_loop_handle.notify_on_permanent_error(error);
-                break;
+                sync_status_loop_handle.notify_on_permanent_error(&err);
+                Timer::sleep(args.error_sleep).await;
+
+                return Err(err.to_string());
             },
         };
 
-        // Validate retrieved block headers.
-        if let Err(err) = validate_headers(ticker, from_block_height, block_headers, storage, &spv_conf).await {
-            error!("Error {err:?} on validating the latest headers for {ticker}!");
-            // Todo: remove this electrum server and use another in this case since the headers from this server are invalid
-            sync_status_loop_handle.notify_on_permanent_error(err);
-            break;
+        return Ok((block_registry, block_headers));
+    }
+
+    let err =
+        format!("Error on retrieving latest {ticker} headers from rpc after {FETCH_BLOCK_HEADERS_ATTEMPTS} attempts");
+    error!("{err}");
+    sync_status_loop_handle.notify_on_permanent_error(&err);
+    Timer::sleep(args.error_sleep).await;
+    Err(err)
+}
+
+/// Retrieves block headers from the specified client within the given height range and revalidate against [`SPVError::ParentHashMismatch`] .
+///
+/// If the height is equal to the starting block height, the loop will break.
+/// Otherwise, it will attempt to fetch the block headers up to the specified `to_height`
+/// It will continue to attempt fetching headers until there's no [`SPVError::ParentHashMismatch`] or runs out of
+/// attempts.
+///
+/// If there is an error during header retrieval, it will check the error type and
+/// take appropriate action based on whether it is a temporary or permanent error.
+///
+async fn retrieve_and_revalidate_mismatching_header(
+    coin: &str,
+    client: &ElectrumClient,
+    mut from_height: u64,
+    storage: &dyn BlockHeaderStorageOps,
+    spv_conf: &SPVConf,
+    sync_status_loop_handle: &mut UtxoSyncStatusLoopHandle,
+) -> Result<(), String> {
+    let args = BlockHeaderUtxoLoopExtraArgs::default();
+    let mut chunk_size = args.chunk_size;
+
+    loop {
+        let to_height = from_height + chunk_size;
+        if from_height == spv_conf.starting_block_header.height {
+            let err = format!("Parent hash mismatch detected for {coin} at height: {from_height}");
```

Comment:
Please give a more informative error message here and when other permanent errors are returned, this is where this electrum is considered bad and we use another.